ARCs/arc-0056.md

```diff
@@ -161,20 +159,32 @@ interface Method {
   desc?: string;
   /** The arguments of the method, in order */
   args: Array<{
-    /** The type of the argument */
+    /** The type of the argument. The `struct` field should also be checked to determine if this arg is a struct. */
     type: ABIType;
     /** If the type is a struct, the name of the struct */
     struct?: StructName;
     /** Optional, user-friendly name for the argument */
     name?: string;
     /** Optional, user-friendly description for the argument */
     desc?: string;
-    /** The default value that clients should use. MUST be base64 encoded bytes */
-    defaultValue?: string;
+    /** The default value that clients should use. */
+    defaultValue?: {
+      /** Base64 encoded bytes or uint64 */
+      data: string | bigint;
```

Comment:
For JavaScript this will need to be `number` to not break deserialisation?

Reply:
I'm wondering if it's just easier to base64 encode uint64 rather than having the number? I could see this causing problems wiht JS otherwise.

Reply:
Yeah I guess that's consistent? We'd have to specify how an `AVMUint64` would be encoded though (maybe we use ARC4 Uint64 encoding in that case so we can reuse `ABIType.from("uint64").decode(value)` rather than a custom decoding function?

Reply:
Yeah exactly I'm thinking we just use ARC4 encoding

Reply:
The jsdoc should be explicit about that

Reply:
It should also mention that `data` will be a UTF8 encoded ARC4 signature when source is `method`

Additionally what should `type` be in the case? Perhaps it should be optional and only for when source is  not `method`?

Reply:
Let me know how 20369ed9c82bc5aa00ae06985d4d24eb3f147861 looks

Reply:
Looks good!